src/components/KeyboardShortcuts/index.tsx

```diff
@@ -142,9 +199,26 @@ export class KeyboardShortcutsBase extends React.Component<Props> {
 
 const mapStateToProps = (
   state: ApplicationState,
-  ownProps: PublicProps,
+  ownProps: PublicProps & RouteComponentProps<PropsFromRouter>,
 ): PropsFromState => {
-  const { versionId } = ownProps;
+  const { history, location, match, versionId } = ownProps;
+
+  const addonId = parseInt(match.params.addonId, 10);
+  const baseVersionId = parseInt(match.params.baseVersionId, 10);
+  const headVersionId = parseInt(match.params.headVersionId, 10);
+  const path = getPathFromQueryString(history) || undefined;
+
+  let compareInfo = null;
+
+  if (addonId && baseVersionId && headVersionId) {
+    compareInfo = getCompareInfo(
```

Comment:
Mapping this object here has a few problems:

* It introduces a dependency on `compareInfo` without any way to load it.
* It copies/pastes all the URL matching logic [from `Compare`](https://github.com/mozilla/addons-code-manager/blob/5ab911bda04346c6d5e9583f467992f9051b86ce/src/pages/Compare/index.tsx#L171-L174) which is fragile, especially since it would be unclear that changing those route params will break this component

I know that Redux is often nicer than prop-drilling but I think we should prop-drill for this case unless there is a necessity to use Redux.

The prop-drilling would only need to go two levels deep. You could render `<VersionFileViewer compareInfo={compareInfo} ... />` from [`Compare`](https://github.com/mozilla/addons-code-manager/blob/5ab911bda04346c6d5e9583f467992f9051b86ce/src/pages/Compare/index.tsx#L140) and render `<KeyboardShortcuts compareInfo={compareInfo} ... />` from [`VersionFileViewer`](https://github.com/mozilla/addons-code-manager/blob/5ab911bda04346c6d5e9583f467992f9051b86ce/src/components/VersionFileViewer/index.tsx#L61).

Also, this would simplify your test setup in some cases.

Reply:
This is a good idea, thanks. I'll implement this.